FABulous/FABulous_CLI/helper.py

```diff
@@ -470,3 +466,147 @@ def execute(self) -> bool:
+                if result.returncode != 0:
+                    logger.warning(f"Git pull failed: {result.stderr}")
+                    logger.info("Attempting fresh clone...")
+                    shutil.rmtree(target_dir)
+                else:
+                    logger.info("✓ Repository updated successfully")
+                    return True
+            else:
+                logger.error(
+                    f"Target directory {target_dir} exists but is not a git repository."
+                    " Please remove or specify a different directory.",
+                )
+                return False
+
+        if not target_dir.exists():
+            # Fresh clone
+            logger.info("Cloning repository...")
+            target_dir.parent.mkdir(parents=True, exist_ok=True)
+
+            result = subprocess.run(
+                [
+                    "git",
+                    "clone",
+                    "--branch",
+                    branch,
+                    "--depth",
+                    "1",
+                    repo_url,
+                    str(target_dir),
+                ],
+                capture_output=True,
+                text=True,
+                timeout=120,
+            )
+
+            if result.returncode != 0:
+                logger.error(f"Failed to clone repository: {result.stderr}")
+                return False
+
+            logger.info("✓ Repository cloned successfully")
+            return True
+
+    except subprocess.TimeoutExpired:
+        logger.error("Git operation timed out")
+        return False
+    except Exception as e:  # noqa: BLE001
+        logger.error(f"Failed to download reference projects: {e}")
+        return False
+
+    return False
+
+
+def install_fabulator(install_dir: Path) -> None:
+    """Installs FABulator into the specified directory by downloading the latest release
+    and sets the FABULATOR_ROOT environment variable in the global .env file.
+
+    Parameters
+    ----------
+        install_dir: Path
+            The directory where FABulator will be installed.
+    """
+
+    fabulator_dir = install_dir / "FABulator"
+    repo_url = "https://github.com/FPGA-Research/FABulator.git"
+
+    if not install_dir.exists():
+        logger.info(f"Creating installation directory {install_dir}")
+        install_dir.mkdir(parents=True, exist_ok=True)
+
+    logger.info(f"Installing FABulator in {fabulator_dir.absolute()}")
+
+    # TODO: Update branch to main, when new release available
+    if not clone_git_repo(repo_url, fabulator_dir, "develop"):
+        raise RuntimeError("Failed to install FABulator. Please install manually.")
+
+    if shutil.which("mvn") is None:
+        logger.warning(
+            "Application mvn (Java Maven) not found in PATH."
+            "FABulator may not work correctly."
+        )
+
+    add_var_to_global_env("FAB_FABULATOR_ROOT", str(fabulator_dir.absolute()))
+
+
+def is_application_installed(app_name: str) -> bool:
```

Comment:
I don't think this is used

Reply:
Yes, this was accidentally pushed. Should be fixed now.